.github/workflows/push_dockerhub_dev.yml

```diff
@@ -0,0 +1,36 @@
+name: nf-core Docker push (dev)
+# This builds the docker image and pushes it to DockerHub
+# Runs on nf-core repo releases and push event to 'dev' branch (PR merges)
+on:
+  push:
+    branches:
+      - dev
+
+jobs:
+  push_dockerhub:
+    name: Push new Docker image to Docker Hub (dev)
+    runs-on: ubuntu-latest
+    # Only run for the nf-core repo, for releases and merged PRs
+    if: ${{ github.repository == 'nf-core/tools'}}
+    env:
+      DOCKERHUB_USERNAME: ${{ secrets.DOCKERHUB_USERNAME }}
+      DOCKERHUB_PASS: ${{ secrets.DOCKERHUB_PASS }}
+    steps:
+      - name: Check out tools code
+        uses: actions/checkout@v2
+
+      - name: Build new docker image for tools
+        run: docker build --no-cache . -t nf-core/tools:dev -f tools.Dockerfile
+
+      - name: Push tools Docker image to DockerHub (dev)
+        run: |
+          echo "$DOCKERHUB_PASS" | docker login -u "$DOCKERHUB_USERNAME" --password-stdin
+          docker push nf-core/tools:dev
+
+      - name: Build new base docker image for templates
+        run: docker build --no-cache . -t nf-core/base:dev -f base.Dockerfile
```

Comment:
This works fine, but it might be a nice opportunity to use a GitHub Actions `matrix` - then we don't need to repeat ourselves and the two jobs will run in parallel. We can do this here because the commands are identical except for the image name string.

See docs: https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstrategymatrix

Sorry, should have suggested this before but only just thought of it.